React/Views/ScrollView/RCTScrollView.m

```diff
@@ -868,6 +868,11 @@ - (void)scrollViewWillEndDragging:(UIScrollView *)scrollView withVelocity:(CGPoi
 
     // Pick snap point based on direction and proximity
     CGFloat fractionalIndex = (targetContentOffsetAlongAxis + alignmentOffset) / snapToIntervalF;
+    if (self.disableIntervalMomentum) {
+      // Use current scroll offset to determine the next index to snap to
+      fractionalIndex = (scrollView.contentOffset.x + alignmentOffset) / snapToIntervalF;
```

Comment:
This is ok, but I think the way Android does all this logic is a lot cleaner where `targetContentOffsetAlongAxis` includes the `alignmentOffset`.